SystemView: Minor Segger SysView improvements

rugeGerritsen opened the pull request:
* Fixes compilation error when THREAD_MONITOR is not enabled
* Logs human readable thread names if available

rugeGerritsen committed:
systemview: SystemView tracing requires THREAD_MONITOR

Otherwise it will not be able to send its thread list.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
systemview: Use thread name when available

This may make it easier to debug.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

nashif commented on subsys/tracing/sysview/tracing_sysview.h:
```diff
@@ -29,9 +30,21 @@ void sys_trace_mutex_unlock(struct k_mutex *mutex);
 
 static inline void sys_trace_thread_info(struct k_thread *thread)
 {
+#if IS_ENABLED(CONFIG_THREAD_NAME)
```
if you look at sysview.c, there is similar code doing exactly the same thing, time to put this in a helper function and use it in both places and avoid duplication.

rugeGerritsen replied:
Fixed

rugeGerritsen committed:
systemview: Use common function to get sysview thread name

Use a common function to get the thread name.
There was no necessity in keeping sys_trace_thread_info() inline,

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

chrta commented on subsys/tracing/sysview/sysview.c:
```diff
@@ -105,6 +105,35 @@ void sys_trace_mutex_unlock(struct k_mutex *mutex)
+	const char *tname = k_thread_name_get(thread);
+
+	if (tname != NULL && tname[0] != '\0') {
+		memcpy(name, tname, sizeof(name));
```
The use of `sizeof(name)` looks wrong in this function. Isn't it always 4 on a 32 bit platform? I would expect that THREAD_NAME_LEN is used instead of this instead.

rugeGerritsen replied:
Thanks for pointing out this copy-paste mistake

rugeGerritsen commented:
Rebased to retrigger CI
